### ISSUE TITLE
MacOS fails to compile wxMaxima-19.01.2 with wxWidgets-3.1.2

### ISSUE THREAD

**Issue by tomio-arisaka** [wxMaxima-developers/wxmaxima#1083]
Hello,

I can compile wxMaxima-19.01.2 with wxWidgets-3.0.4 on macOS.
But I cannot compile wxMaxima-19.01.2 with **_wxWidgets-3.1.2_** on macOS.
I'm using MacPorts on the following macOS:
```
$ sw_vers
ProductName:	Mac OS X
ProductVersion:	10.11.6
BuildVersion:	15G22010
$ xcodebuild -version
Xcode 8.2.1
Build version 8C1002
```

I got the following error:

```
:notice:build --->  Building wxMaxima
:debug:build Executing org.macports.build (wxMaxima)
:debug:build system -W /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_math_wxMaxima/wxMaxima/work/wxMaxima-developers-wxmaxima-3ae37ee: mkdir ./build
:debug:build system -W /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_math_wxMaxima/wxMaxima/work/wxMaxima-developers-wxmaxima-3ae37ee/build: cmake -DwxWidgets_CONFIG_EXECUTABLE:FILEPATH=/opt/local/Library/Frameworks/wxWidgets.framework/Versions/wxWidgets/3.1/bin/wx-config ..
:info:build -- The C compiler identification is AppleClang 8.0.0.8000042
:info:build -- The CXX compiler identification is AppleClang 8.0.0.8000042

...
...

:debug:build system -W /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_math_wxMaxima/wxMaxima/work/wxMaxima-developers-wxmaxima-3ae37ee/build: cmake --build .

...
...

:info:build [ 32%] Building CXX object src/CMakeFiles/wxmaxima.dir/ConfigDialogue.cpp.o
:info:build /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_math_wxMaxima/wxMaxima/work/wxMaxima-developers-wxmaxima-3ae37ee/src/ConfigDialogue.cpp:678:60: error: too many arguments to function call, expected 0, have 1
:info:build       m_wxStartupCommands->OSXDisableAllSmartSubstitutions(true);
:info:build       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^~~~
:info:build /opt/local/Library/Frameworks/wxWidgets.framework/Versions/wxWidgets/3.1/include/wx-3.1/wx/osx/textctrl.h:136:5: note: 'OSXDisableAllSmartSubstitutions' declared here
:info:build     void OSXDisableAllSmartSubstitutions();
:info:build     ^
:info:build /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_math_wxMaxima/wxMaxima/work/wxMaxima-developers-wxmaxima-3ae37ee/src/ConfigDialogue.cpp:723:7: error: use of undeclared identifier 'm_StartupCommands'; did you mean 'm_startupCommands'?
:info:build       m_StartupCommands->OSXDisableAllSmartSubstitutions(true);
:info:build       ^~~~~~~~~~~~~~~~~
:info:build       m_startupCommands
:info:build /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_math_wxMaxima/wxMaxima/work/wxMaxima-developers-wxmaxima-3ae37ee/src/ConfigDialogue.h:245:15: note: 'm_startupCommands' declared here
:info:build   wxTextCtrl *m_startupCommands;
:info:build               ^
:info:build /opt/local/var/macports/build/_opt_local_var_macports_sources_rsync.macports.org_macports_release_tarballs_ports_math_wxMaxima/wxMaxima/work/wxMaxima-developers-wxmaxima-3ae37ee/src/ConfigDialogue.cpp:723:58: error: too many arguments to function call, expected 0, have 1
:info:build       m_StartupCommands->OSXDisableAllSmartSubstitutions(true);
:info:build       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^~~~
:info:build /opt/local/Library/Frameworks/wxWidgets.framework/Versions/wxWidgets/3.1/include/wx-3.1/wx/osx/textctrl.h:136:5: note: 'OSXDisableAllSmartSubstitutions' declared here
:info:build     void OSXDisableAllSmartSubstitutions();
:info:build     ^
:info:build 3 errors generated.
:info:build make[2]: *** [src/CMakeFiles/wxmaxima.dir/ConfigDialogue.cpp.o] Error 1
:info:build make[1]: *** [src/CMakeFiles/wxmaxima.dir/all] Error 2
:info:build make: *** [all] Error 2
:info:build Command failed: cmake --build .
:info:build Exit code: 2
:error:build Failed to build wxMaxima: command execution failed
```
So I tried to make a patch to avoid the issue as follows:

```
$ diff -u ./src/ConfigDialogue.cpp.orig ./src/ConfigDialogue.cpp
--- ./src/ConfigDialogue.cpp.orig	2019-01-22 18:48:06.000000000 +0900
+++ ./src/ConfigDialogue.cpp	2019-01-26 21:37:14.000000000 +0900
@@ -675,7 +675,7 @@
                                      wxTE_MULTILINE | wxHSCROLL);
   #ifdef __WXMAC__
     #if wxCHECK_VERSION(3, 1, 1)
-      m_wxStartupCommands->OSXDisableAllSmartSubstitutions(true);
+      m_wxStartupCommands->OSXDisableAllSmartSubstitutions();
     #endif
   #endif
   m_wxStartupCommands->SetValue(contents);
@@ -720,7 +720,7 @@
   m_startupCommands->SetValue(contents);
   #ifdef __WXMAC__
     #if wxCHECK_VERSION(3, 1, 1)
-      m_StartupCommands->OSXDisableAllSmartSubstitutions(true);
+      m_startupCommands->OSXDisableAllSmartSubstitutions();
     #endif
   #endif
 
$ 
```

But I don't know whether the patch is correct or not.
Because I'm not familiar with wxWidgets.

Please fix the issue.
Thanks.

**Comment by tomio-arisaka** [comment]
Thanks for the comment!

I looked at the commit 204bdda24f1ca142edf3f29bbc40efd40b64bfc2.
I think you have missed "**_m_StartupCommands_**".
The error message explained that the cause of the error is not only the "**_true_**" argument but also the "**_m_StartupCommands_**" variable.
The third letter of "**_m_StartupCommands_**" is "**_S_**" which is the uppercase letter. But should it be the lowercase letter?

By the way, when compiling wxMaxima, Travis-CI uses "wxmac-3.0.4" (wxWidgets-3.0.4).
So Travis-CI does not check the source code in order to link wxWidgets-3.1.2.

Thanks,
Tomio Arisaka

**Comment by gunterkoenigsmann** [comment]
Of course you are right! Thanks a lot!

Hope wxMaxima compiles now. In theory all multi-window and display problems should be resolved by now (except wxWidgets coloring gui items wrong in dark mode, that is). But as I have no mac computer my theory might be entirely wrong...

**Comment by gunterkoenigsmann** [comment]
Tried to make travis-ci compile wxMaxima using the newest wxWidgets version. But it seems that there is no possibility to to so, currently.

**Comment by tomio-arisaka** [comment]
Thanks a lot!

I checked the commit 1dc29273812aeec8b74d5bc973edf22e2a9fa3ba,
then compiling it with wxWidgets-3.1.2 was successful.
It seems that compiled wxMaxima does not have multi-window and display problems.

But my MacBook is not supported by macOS-Mojave, so I don't know the dark-mode issue.


About wxWidgets-3.1.2,

I think the current wxmac of homebrew does not support the development version of wxWidgets.
https://github.com/Homebrew/homebrew-core/blob/master/Formula/wxmac.rb

For some reason, the development version (3.1.1) has been removed.
https://discourse.brew.sh/t/brew-install-wxwidgets-devel-3-1-1/3453
